src/learning/gp.rs

```diff
@@ -193,11 +203,11 @@ impl<T: Kernel, U: MeanFunc> GaussianProcess<T, U> {
 
             let v_mat = Matrix::new(test_mat.rows(), test_mat.cols(), var_data);
 
-            let post_var = self.ker_mat(inputs, inputs) - &v_mat * v_mat.transpose();
+            let post_var = try!(self.ker_mat(inputs, inputs)) - &v_mat * v_mat.transpose();
 
-            return (post_mean, post_var);
+            Ok((post_mean, post_var))
+        } else {
+            Err(Error::new(ErrorKind::UntrainedModel, "The model has not been trained."))
```

Comment:
This is really happening a lot. Can't we have a special constructor or a more advanced `ErrorKind` that has or not a `&'static str` message?